net/http/partitioned_host_state_map.cc

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/net/http/partitioned_host_state_map.h"
+
+#include <utility>
+
+#include "base/strings/strcat.h"
+#include "crypto/sha2.h"
+#include "net/base/network_isolation_key.h"
+
+namespace net {
+
+PartitionedHostStateMapBase::PartitionedHostStateMapBase() = default;
+PartitionedHostStateMapBase::~PartitionedHostStateMapBase() = default;
+
+base::AutoReset<absl::optional<std::string>>
+PartitionedHostStateMapBase::SetPartitionHash(
+    absl::optional<std::string> partition_hash) {
+  CHECK(!partition_hash || partition_hash->empty() ||
+        partition_hash->size() == crypto::kSHA256Length);
+  return base::AutoReset<absl::optional<std::string>>(
+      &partition_hash_, std::move(partition_hash));
+}
+
+bool PartitionedHostStateMapBase::HasPartitionHash() const {
+  return partition_hash_.has_value();
+}
+
+bool PartitionedHostStateMapBase::IsPartitionHashValid() const {
+  return partition_hash_ && !partition_hash_->empty();
+}
+
+std::string PartitionedHostStateMapBase::GetKeyWithPartitionHash(
+    const std::string& k) const {
+  CHECK(IsPartitionHashValid());
+  return base::StrCat({GetHalfKey(k), GetHalfKey(*partition_hash_)});
```

Comment:
@kdenhartog plz take a look

Reply:
As a note of record for anyone who might look at this review later. We discussed this in person because we thought it might be susceptible to a crafted URL attack to build cross site access to a partitioned cache. After looking at this further we didn't believe there was an actual issue here.